pkg/asset/manifests/dns.go

```diff
@@ -108,11 +109,14 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 		}
 	case gcptypes.Name:
 		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			zone, err := icgcp.GetPublicZone(context.TODO(), installConfig.Config.Platform.GCP.ProjectID, installConfig.Config.BaseDomain)
-			if err != nil {
-				return errors.Wrapf(err, "failed to get public zone for %q", installConfig.Config.BaseDomain)
+			if installConfig.Config.GCP.PublicZoneID == "" {
+				zone, err := icgcp.GetPublicZone(context.TODO(), installConfig.Config.Platform.GCP.ProjectID, installConfig.Config.BaseDomain)
+				if err != nil {
+					return errors.Wrapf(err, "failed to get public zone for %q", installConfig.Config.BaseDomain)
+				}
+				installConfig.Config.GCP.PublicZoneID = strconv.FormatUint(zone.Id, 10)
```

Comment:
this is _fragile_ if you think it will set the installconfig asset for later too, the installconfig is a dependency for this asset and the store (asset.Store) could have easily provided a copy of it.

Reply:
```go
publicZoneID := installConfig.Config.GCP.PublicZoneID
if publicZoneID == "" {
    zone, err := Get
   ....
    publicZoneID = zone.Name
}
config.Spec.PublicZone = &configv1.DNSZone{ID: publicZoneID}
```
something like that should be enough.

Reply:
I'm not following, can you elaborate on the issue here?

Reply:
?